debezium-server/debezium-server-core/src/main/java/io/debezium/server/storage/JdbcDatabaseHistory.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        lock.write(() -> {
+            if (running.compareAndSet(false, true)) {
+                if (conn == null) {
+                    throw new IllegalStateException("Database connection must be set before it is started");
+                }
+                try {
+                    if (!storageExists()) {
+                        initializeStorage();
+                    }
+                }
+                catch (Exception e) {
+                    throw new DatabaseHistoryException("Unable to create history table " + jdbc_uri + ": " + e.getMessage(), e);
+                }
+            }
+        });
+    }
+
+    @Override
+    protected void storeRecord(HistoryRecord record) throws DatabaseHistoryException {
+        if (record == null) {
+            return;
+        }
+        lock.write(() -> {
+            if (!running.get()) {
+                throw new IllegalStateException("The history has been stopped and will not accept more records");
+            }
+
+            try {
+                logger.error("Inserting records into the table...");
+                String line = writer.write(record.document());
+                String currentTimeMillis = Long.toString(System.currentTimeMillis());
+                List<String> substrings = Splitter.fixedLength(65000).splitToList(line);
+                if (substrings.size() >= 4) {
+                    throw new DatabaseHistoryException("Failed to save history data!");
+                }
+                PreparedStatement sql = conn.prepareStatement("INSERT INTO " + DATABASE_HISTORY_TABLE_NAME + " VALUES ( ?, ?, ?, ? )");
+                sql.setString(1, currentTimeMillis);
+                sql.setString(2, substrings.size() >= 1 ? substrings.get(0) : "");
+                sql.setString(3, substrings.size() >= 2 ? substrings.get(1) : "");
+                sql.setString(4, substrings.size() >= 3 ? substrings.get(2) : "");
+                sql.executeUpdate();
+                // todo check length 4 and above very long ones!
+            }
+            catch (IOException | SQLException e) {
+                logger.error("Failed to convert record to string: {}", record, e);
+            }
+        });
+    }
+
+    @Override
+    public void stop() {
+        running.set(false);
+        super.stop();
+        try {
+            conn.close();
+        }
+        catch (SQLException e) {
+            logger.error("Exception during stop", e);
+        }
+    }
+
+    @Override
+    protected synchronized void recoverRecords(Consumer<HistoryRecord> records) {
+        lock.write(() -> {
+            try {
+                if (exists()) {
+                    Statement stmt = conn.createStatement();
+                    ResultSet rs = stmt.executeQuery(
+                            "SELECT history_insert_epoch_ms, history_data_part1, history_data_part2, history_data_part3 FROM " + DATABASE_HISTORY_TABLE_NAME);
```

Comment:
Should be ordered?! Thinking about it, you'd have to add some sequence number too, so to resolve conflicts between entries with the same timestamp.